app/models/katello/deb.rb

```diff
@@ -103,8 +103,13 @@ def self.apt_installable_for_host(host)
       Katello::Deb.in_repositories(repos).where.not(name: host.installed_debs.pluck(:name)).order(:name)
     end
 
-    def self.latest(_relation)
-      fail 'NotImplemented'
+    def self.latest(relation)
+      # This might be very slow
+      return relation.joins(
+        "LEFT OUTER JOIN(#{relation.to_sql}) AS katello_debs2 ON " \
+        'katello_debs.name = katello_debs2.name AND katello_debs.architecture = katello_debs2.architecture AND ' \
+        'deb_version_cmp(katello_debs.version, katello_debs2.version) < 0 ' \
+      ).where('katello_debs2.id IS NULL')
```

Comment:
Capturing things we talked about live: If I were doing this I'd probably drop the explicit return and used a squiggly heredoc, but that's just personal preference so feel free to ignore.

Reply:
Since most of the code was shamelessly copied over from app/models/katello/rpm.rb, I would keep it that way for being consistent.

Eventually, we should think about enforcing a certain style through rubocop. AFAIK Katello's rubocop rules are currently pretty flexible.